Source/Renderer/FramebufferManager.js

```diff
@@ -0,0 +1,308 @@
+    );
+  }
+  //>>includeEnd('debug');
+  depthTexture = defaultValue(depthTexture, false);
+  hdr = defaultValue(hdr, false);
+
+  if (this._attachmentsSet || this.isDirty(width, height, hdr)) {
+    if (!this._attachmentsSet) {
+      this.destroyResources();
+    }
+    this._useHdr = hdr;
+
+    // Create color texture
+    if (this._color && this._createColorAttachments) {
+      var pixelDatatype = hdr
+        ? context.halfFloatingPointTexture
+          ? PixelDatatype.HALF_FLOAT
+          : PixelDatatype.FLOAT
+        : PixelDatatype.UNSIGNED_BYTE;
+      for (var i = 0; i < this._colorAttachmentsLength; ++i) {
+        this._colorTextures.push(
+          new Texture({
+            context: context,
+            width: width,
+            height: height,
+            pixelFormat: PixelFormat.RGBA,
+            pixelDatatype: pixelDatatype,
+            sampler: Sampler.NEAREST,
+          })
+        );
+      }
+    }
+
+    // Create depth stencil texture or renderbuffer
+    if (this._depthStencil && this._createDepthAttachments) {
+      if (depthTexture) {
+        this._depthStencilTexture = new Texture({
+          context: context,
+          width: width,
+          height: height,
+          pixelFormat: PixelFormat.DEPTH_STENCIL,
+          pixelDatatype: PixelDatatype.UNSIGNED_INT_24_8,
+          sampler: Sampler.NEAREST,
+        });
+      } else {
+        this._depthStencilRenderbuffer = new Renderbuffer({
+          context: context,
+          width: width,
+          height: height,
+          format: RenderbufferFormat.DEPTH_STENCIL,
+        });
+      }
+    }
+
+    // Create depth texture
+    if (this._depth && this._createDepthAttachments) {
+      if (depthTexture) {
+        this._depthTexture = new Texture({
+          context: context,
+          width: width,
+          height: height,
+          pixelFormat: PixelFormat.DEPTH_COMPONENT,
+          pixelDatatype: PixelDatatype.UNSIGNED_INT,
+          sampler: Sampler.NEAREST,
+        });
+      } else {
+        this._depthRenderbuffer = new Renderbuffer({
+          context: context,
+          width: width,
+          height: height,
+          format: RenderbufferFormat.DEPTH_COMPONENT16,
+        });
+      }
+    }
+
+    this._framebuffer = new Framebuffer({
+      context: context,
+      colorTextures: this._colorTextures,
+      depthTexture: this._depthTexture,
+      depthRenderbuffer: this._depthRenderbuffer,
+      depthStencilTexture: this._depthStencilTexture,
+      depthStencilRenderbuffer: this._depthStencilRenderbuffer,
+      destroyAttachments: false,
+    });
+    this._attachmentsSet = false;
+  }
+};
+
+FramebufferManager.prototype.getColorTexture = function (index) {
+  index = defaultValue(index, 0);
+  return this._colorTextures[index];
+};
+
+FramebufferManager.prototype.setColorTexture = function (texture, index) {
+  //>>includeStart('debug', pragmas.debug);
+  if (this._createColorAttachments) {
```

Comment:
Is it safe to allow this function to work on any index? The `Framebuffer` equivalent is much stricter.